### PR TITLE
Exit confirmation dialog

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game/Screens/Menu/ExitConfirmDialog.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace osu.Game.Screens.Select
+{
+    public class ExitConfirmDialog : PopupDialog
+    {
+        public Action OnExit;
+        public ExitConfirmDialog()
+        {
+            Icon = FontAwesome.fa_question;
+            HeaderText = @"Confirm exit";
+            BodyText = @"You'll left";
```
Comment (Jorolf) [inline osu.Game/Screens/Menu/ExitConfirmDialog.cs]: You'll leave

Comment (EVAST9919) [inline osu.Game/Screens/Menu/ExitConfirmDialog.cs]: That was just a random text

File: osu.Game/Screens/Menu/MainMenu.cs [hunk 2]
```diff
@@ -117,14 +126,34 @@ protected override bool OnExiting(Screen next)
...
+                });
+            }
+            else Exit();
+        }
+
         protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
         {
+            if (args.Repeat) return false;
+
             if (!args.Repeat && state.Keyboard.ControlPressed && state.Keyboard.ShiftPressed && args.Key == Key.D)
```
Comment (Jorolf) [inline osu.Game/Screens/Menu/MainMenu.cs]: If you're going to add a seperate if statement for testing if a key press was repeated you can remove this here

File: osu.Game/Screens/Menu/MainMenu.cs [hunk 3]
```diff
@@ -117,14 +126,34 @@ protected override bool OnExiting(Screen next)
             return base.OnExiting(next);
         }
 
+        private void promptExit()
+        {
+            if (confirmExit)
+            {
+                dialog?.Push(new ExitConfirmDialog()
```
Comment (peppy) [inline osu.Game/Screens/Menu/MainMenu.cs]: this is wrong. you're doing a null check but in the case it fails what happens? nothing.

File: osu.Game/Screens/Menu/MainMenu.cs [hunk 4]
```diff
@@ -47,20 +52,24 @@ public MainMenu()
...
-        private void load(OsuGame game)
+        private void load(OsuGame game, DialogOverlay dialogOverlay, OsuConfigManager config)
         {
             LoadComponentAsync(background);
 
+            confirmExit = config.GetBindable<bool>(OsuConfig.ConfirmExit);
+
             buttons.OnSettings = game.ToggleOptions;
 
+            dialog = dialogOverlay;
```
Comment (peppy) [inline osu.Game/Screens/Menu/MainMenu.cs]: don't call the same variable different things.

`this.dialog = dialog`

**Comment by peppy** [comment]
- this doesn't cover alt-f4, only escape
- this only covers exiting at the main menu. what about exiting from another screen?

pretty sure you'll want to look into `OnExiting` callbacks and not handle this local to the main menu.

**Review by peppy — CHANGES_REQUESTED** [review]
as commented

**Pull request closed without merging**